### PR TITLE
Fix hot reloading

### PR DESCRIPTION
**Pull request by Demonthos**
Reimplements hot reloading

todo:
- [x] create trait for namespaces to implement that allows for hot reloading outside of the default html namespace
- [x] build Template<'staitc> from CallBody
- [x] create helper to diff CallBodys
- [x] web
- [x] desktop
- [x] companion PR to integrate with cli

### CODE REVIEW

**Comment by jkelleyrtp** [comment]
Much appreciated! Thank you!

File: packages/html/src/global_attributes.rs [hunk 1]
```diff
@@ -25,2273 +71,2275 @@ macro_rules! trait_methods {
...
-pub trait GlobalAttributes {
-    trait_methods! {
-        @base
 
-        /// Prevent the default action for this element.
-        ///
-        /// For more information, see the MDN docs:
-        /// <https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault>
-        prevent_default: "dioxus-prevent-default";
+trait_methods! {
```
Comment (jkelleyrtp) [inline packages/html/src/global_attributes.rs:75]: Can we keep it so we're running the impls inside the trait declared outside the macro?

This should keep the diff smaller and be a little bit more obvious from the outside what's happening.

Comment (Demonthos) [reply packages/html/src/global_attributes.rs:75]: We could do that but then the map_global_attributes and map_svg_attributes functions would be associated with the trait

Comment (jkelleyrtp) [reply packages/html/src/global_attributes.rs:75]: Oh, I understand now. Sorry about that, I will merge an equivalent to master so the diff goes down.

**Comment by jkelleyrtp** [comment]
Gonna play with some of the organization a bit to cut down on dependency chains, but looks good! I hope to follow this up with a PR that turns on hot reloading for desktop by default.

**Review by jkelleyrtp — APPROVED** [review]

**Pull request merged**